### ISSUE TITLE
Fails with 2 transform streams and lot of data

### ISSUE THREAD

**Issue by anton-makarov** [deoxxa/duplexer2#23]
Below code will fail with error RangeError: Maximum call stack size exceeded.

Node version: 0.10.37

Failed code:

``` js
'use strict';

var TransformStream = require('stream').Transform,
	util = require('util'),
	duplexer = require('duplexer2');

var MyTransformStream = function() {
	TransformStream.call(this);
};
util.inherits(MyTransformStream, TransformStream);

MyTransformStream.prototype._transform = function(chunk, encoding, callback) {
	callback(null, chunk);
};

var firstStream = new MyTransformStream(),
	secondStream = new MyTransformStream();

firstStream.pipe(secondStream);

var duplexStream = duplexer({
	bubbleErrors: false
}, firstStream, secondStream);

for (var i = 0; i < 3000; i++) {
	duplexStream.write(i.toString());
}
```

**Comment by okv** [comment]
same for me with nodejs 0.10.42: 
```
(node) warning: Recursive process.nextTick detected. This will break in the next version of node. Please use setImmediate for recursive deferral.
(node) warning: Recursive process.nextTick detected. This will break in the next version of node. Please use setImmediate for recursive deferral.
....
RangeError: Maximum call stack size exceeded
```
it crashes whole process.

It works without error on higher versions of node (i've tried 0,12 and 5.5).

**Comment by juliangruber** [comment]
this error doesn't happen starting at node 0.12. so i suggest you upgrade your node versions, 0.10 is quite old

**Comment by okv** [comment]
nodejs 0.10 still widely used across different environments (e.g. it's default in openshift service, ubuntu lts, centos, etc). readable-stream used in this lib supports nodejs 0.10 correctly and i think duplexer2 should do so.

**Comment by shinnn** [comment]
@okv Deleted your meaningless [cross-posted](https://github.com/juliangruber/multipipe/issues/30#issuecomment-188271414) comment.

**Comment by okv** [comment]
@shinnn sorry for that, but why you did not remove @juliangruber [cross-posted](https://github.com/juliangruber/multipipe/issues/30#issuecomment-188261585) comment?
@shinnn anyway, what about 0.10 support? Is it supported by deplexer2?

**Comment by shinnn** [comment]
@okv However I, and probably other maintainers, don't have enough interest and time to add a workaround for Node v0.10, we don't decline awesome pull requests to fix the v0.10-specific problem.

**Comment by shinnn** [comment]
> why you did not remove @juliangruber cross-posted comment?

Ah, sorry I didn't notice that. Now deleted both.

**Comment by okv** [comment]
@shinnn looks like that bug related to readable stream. Is readable stream supposed to work with 0.10? it's goal (support stream 2 and 3 for earlier versions of node including 0.10) of readable stream, isn't it?

**Comment by shinnn** [comment]
> Is readable stream supposed to work with 0.10? 

https://github.com/nodejs/readable-stream/issues/new

**Comment by deoxxa** [comment]
```js
'use strict';

var TransformStream = require('stream').Transform,
    util = require('util'),
    duplexer = require('duplexer2');

var MyTransformStream = function() {
  TransformStream.call(this);
};
util.inherits(MyTransformStream, TransformStream);

MyTransformStream.prototype._transform = function(chunk, encoding, callback) {
  callback(null, chunk);
};

var writable = new MyTransformStream(),
    readable = new MyTransformStream();

writable.pipe(readable);

var duplexStream = duplexer({
  bubbleErrors: false
}, writable, readable);

for (var i = 0; i < 3000; i++) {
  duplexStream.write(i.toString());
}
```

I made two small changes to your code - I renamed `firstStream` to `writable`, and `secondStream` to `readable`. Why are you piping your writable stream into your readable stream?

**Comment by anton-makarov** [comment]
@deoxxa this is just example of how to get this error, not actual usage. Found out that this error happens because of readable-stream, added issue there: https://github.com/nodejs/readable-stream/issues/187

**Comment by okv** [comment]
FYI: linked readable stream issue fixed in 2.0.6, but duplexer 2 (with readable stream 2.0.6) still fails with same error.
